src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -11,9 +11,81 @@
  * Реализуйте метод moveRobot()</p>
  */
 public class FieldXY {
-
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        moveX(robot, toX);
+        moveY(robot, toY);
+    }
+
+    private void moveX(Robot robot, int toX) {
+        while (robot.getX() != toX) {
+            switch (robot.getDirection()) {
+                case UP:
+                    if (robot.getX() > toX) {
+                        robot.turnLeft();
+                    } else {
+                        robot.turnRight();
+                    }
+                    break;
+                case DOWN:
+                    if (robot.getX() < toX) {
+                        robot.turnLeft();
+                    } else {
+                        robot.turnRight();
+                    }
+                    break;
+                case LEFT:
+                    if (robot.getX() < toX) {
+                        robot.turnLeft();
+                        robot.turnLeft();
```

Comment:
Try to optimize this method. What is the difference in logic between this method and moveY?Can we combine them?

Reply:
done